### PR TITLE
docs: update state machine description in readme

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Updates the README to reflect the current state of the Updater state machines.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: library/README.md [hunk 1]
```diff
@@ -119,35 +125,90 @@ NDK_HOME=$HOME/Documents/GitHub/engine/src/third_party/android_tools/ndk cargo n
...
+- last_attempted_patch: The last patch that we attempted to boot.
+- next_boot_patch: The next patch that we will attempt to boot.
+
+This state machine has the following states. It can only move forward through
+them.
+
+1. Ready - The engine is initialized but has not started booting yet.
+2. Booting - The engine has started booting.
+3. Success - The engine successfully booted.
+4. Failure - The engine failed to boot.
```
Comment (eseidel) [inline library/README.md:151]: Is this actually recorded in an enum?  (if not that might be a next step, or worth commenting).

File: library/README.md [hunk 2]
```diff
@@ -119,35 +125,90 @@ NDK_HOME=$HOME/Documents/GitHub/engine/src/third_party/android_tools/ndk cargo n
...
+- `record_launch_failure`: Moves from Booting to Failure
+  - next_boot_patch artifacts are deleted.
+  - next_boot_patch is set to either:
+    - last_boot_patch if it is still valid, or
+    - None, if last_boot_patch is None or invalid.
+
+These are effectively no-ops if we are not booting from a patch.
+
+We assume that this state machine will have advanced at least as far as the
+Booting state before the Patch Check State Machine (below) is started.
```
Comment (eseidel) [inline library/README.md:171]: Is any call to an earlier method an error?  Should it be?

**Review by eseidel — APPROVED** [review]
This is fantastic!  I think the code should just do this now and would be more clear. :)

File: library/README.md (hunk 1 above)
Comment (bryanoltman) [reply library/README.md:151]: Added an assumptions list at the bottom and specifically called out that PatchManager state is implicit, but should be more explicit.

File: library/README.md (hunk 2 above)
Comment (bryanoltman) [reply library/README.md]: This also became part of the "not enforced but should be" assumptions list above.

**Comment by bryanoltman** [comment]
> I think the code should just do this now and would be more clear

It does :) I think the fact that _that_ isn't clear should probably be addressed though 🙃

**Pull request merged**